pinecone/client.go

```diff
@@ -210,6 +582,51 @@ func (c *Client) CreatePodIndex(ctx context.Context, in *CreatePodIndexRequest)
 	return decodeIndex(res.Body)
 }
 
+// CreateServerlessIndexRequest holds the parameters for creating a new [Serverless] index.
+//
+// Fields:
+//   - Name: The name of the Serverless index.
+//   - Dimension: The dimension of the index (must match dimensionality of upserted vectors).
+//   - Metric: The metric used to measure the [similarity] between vectors.
+//   - Cloud: The [cloud provider] in which the index will be created.
```

Comment:
Same feedback regarding the other index fields in the pods-based approach, maybe add a bit more specificity. You can pull directly from the OAS spec: https://github.com/pinecone-io/apis/blob/fbd9d8d48c95b12c6b28aea4e496b6f28666d2d1/src/release/control/resources/indexes/models/ServerlessSpec.yaml#L14